server/channels/db/migrations/postgres/000109_external_user_id.up.sql

```diff
@@ -0,0 +1,2 @@
+ALTER TABLE users ADD COLUMN IF NOT EXISTS externaluserid varchar(255);
```

Comment:
Hey @jespino - since this PR adds new migrations, we will need to test this migration on a large dataset and report back the time it took. You can just take the data dump available in the developers-performance channel (for both MySQL and Postgres), import it, and run the migration yourself.

Let me know if you would be up for that. All of this is due to stricter guidelines around schema changes.

Reply:
That sounds sensible, for sure I can do it.

Reply:
This is instantaneous for posgres, now I going to try with MySQL
```
$ time morph apply up --driver postgres --dsn "postgres://xxxxx:xxxxx@localhost:5432/test-migration?sslmode=disable" --path ./channels/db/migrations/postgres --number 1 -t 360
Attempting to apply 1 migrations...
2023/05/15 16:20:51 ==  external_user_id: migrating  ===================================================================
2023/05/15 16:20:51 ==  external_user_id: migrated (0.0147s)  ==========================================================
1 migrations applied.
morph apply up --driver postgres --dsn  --path  --number 1 -t 360  0,00s user 0,01s system 10% cpu 0,081 total
```

Reply:
Yep, MySQL is where the challenge is :)

Reply:
I'm changing the migration, because we are no longer using a new field, we are expanding the existing one:

Here is the postgres result:
```
$time morph apply up --driver postgres --dsn "postgres://mmuser:mostest@localhost:5432/test-migration?sslmode=disable" --path ./channels/db/migrations/postgres --number 1 -t 360
Attempting to apply 1 migrations...
2023/05/16 16:08:35 ==  increase_user_remote_id_field_size: migrating  =================================================
2023/05/16 16:08:35 ==  increase_user_remote_id_field_size: migrated (0.0107s)  ========================================
1 migrations applied.
morph apply up --driver postgres --dsn  --path  --number 1 -t 360  0,00s user 0,01s system 12% cpu 0,094 total
```

Again, MySQL still pending.

Reply:
Now with mysql:
```
$ time morph apply up --driver mysql --dsn "mmuser:mostest@/mattermost_test?multiStatements=true" --path ./channels/db/migrations/mysql --number 1 -t 360
Attempting to apply 1 migrations...
2023/05/16 16:48:27 ==  increase_user_remote_id_field_size: migrating  =================================================
2023/05/16 16:48:33 ==  increase_user_remote_id_field_size: migrated (6.2264s)  ========================================
1 migrations applied.
morph apply up --driver mysql --dsn  --path ./channels/db/migrations/mysql  1  0,00s user 0,01s system 0% cpu 6,338 total
```

It took 6s for 14087 users. I guess that is acceptable.

Reply:
Ahh you are altering the column type. Unfortunately, that is the _only_ migration that takes table locks. What we are trying to achieve with the new [guidelines](https://docs.google.com/document/d/1wdQLzS_C6cjJIcyZWEYEeaP3Ej_NuuXj4lDt1VFuyZU/edit#) is that NO migration should take table locks at all.

I'm afraid, but we cannot simply add "ALTER COLUMN" migration any more. Any other migration (like adding a new column) is completely fine. But if you _must_ do this, then there are certain set of "steps" that needs to be done: https://docs.google.com/document/d/1wdQLzS_C6cjJIcyZWEYEeaP3Ej_NuuXj4lDt1VFuyZU/edit#heading=h.f6dz13s1pdbr to achieve this without downtime.

Essentially:

- Create a new column
- Migrate existing data
- From next ESR, start using the new column.
- Next ESR, drop the old column.

tldr is: altering a column locks the table. We cannot have any migration acquire table locks from now on.

Reply:
Removing the migration entirely. This is not strictly needed for the purpose of this PR. It was handy to get it solved right away, but we can solve it as a different thing, in a different PR.